features/netsocket/DNS.h

```diff
@@ -72,14 +72,15 @@ class DNS {
      *
      *  @param host     Hostname to resolve.
      *  @param callback Callback that is called to return the result.
+     *  @param interface_name  Network interface name
      *  @param version  IP version of address to resolve. NSAPI_UNSPEC indicates that the
      *                  version is chosen by the stack (defaults to NSAPI_UNSPEC).
      *  @return         NSAPI_ERROR_OK on immediate success,
      *                  negative error code on immediate failure or
      *                  a positive unique ID that represents the hostname translation operation
      *                  and can be passed to cancel.
      */
-    virtual nsapi_value_or_error_t gethostbyname_async(const char *host, hostbyname_cb_t callback,
+    virtual nsapi_value_or_error_t gethostbyname_async(const char *host, hostbyname_cb_t callback, const char *interface_name = NULL,
                                                        nsapi_version_t version = NSAPI_UNSPEC) = 0;
```

Comment:
Is it _really_ necessary to break existing applications? Could there be separate overload of gethostname_async() with interface name (without default value) or at least have the interface_name as last argument, so the existing code would compile without modifications?

Reply:
Or after a second thought... reorder the parameters, so that we add the new parameter to end.

Reply:
Parameters  for gethostbyname reordered. Interface name is now last and is NULL by default